src/bridgestan.h

```diff
@@ -280,6 +282,38 @@ int bs_log_density_hessian(const bs_model* m, bool propto, bool jacobian,
                            const double* theta_unc, double* val, double* grad,
                            double* hessian, char** error_msg);
 
+/**
+ * Calculate the log density and the product of the Hessian with the specified
+ * vector for the specified unconstrain parameters and write it into the
+ * specified value pointer and Hessian-vector product pointer, dropping
+ * constants it `propto` is `true` and including the Jacobian adjustment if
+ * `jacobian` is `true`. Returns a return code of 0 for success
+ * and -1 if there is an exception executing the Stan program. The
+ * pointer `hvp` must have enough space to hold the product.
+ *
+ * Hessian-vector-products are computed using nested automatic
+ * differentiation if BRIDGESTAN_AD_HESSIAN is defined, otherwise
+ * they are computed using central finite differences.
```

Comment:
same question about central finite differences of gradient vectors at theta_unc

Reply:
perterbed -> perturbed

Can this really be done with just two gradient evals?

Reply:
Yes, it only needs two. The best explanation is probably Justin's blog: https://justindomke.wordpress.com/2009/01/17/hessian-vector-products/